### PR TITLE
Bug 1561208 - Clean up react-intl usage (shallowWithIntl, mountWithIntl, injectIntl, addLocaleData, IntlProvider)

### PR DESCRIPTION
**Pull request by JeaneC**
Waiting to rebase ontop of about:welcome changes (#5139). (Look at last commit only).

Cleanup of React-Intl general uses in components and test cases.

### CODE REVIEW

File: locales-src/onboarding.ftl [hunk 1]
```diff
@@ -29,6 +29,29 @@ onboarding-join-form-legal = By proceeding, you agree to the <a data-l10n-name="
...
 onboarding-start-browsing-button-label = Start Browsing
+onboarding-menu-dismiss =
+    .title = Dismiss
+    .aria-label = Dismiss
+
+## Firefox Sync modal dialog strings.
+
+onboarding-sync-welcome-header = Take { -brand-product-name } with You
+onboarding-sync-welcome-content = Get your bookmarks, history, passwords and other settings on all your devices.
+onboarding-sync-welcome-learn-more-link = Learn more about { -brand-product-name } Accounts
```
Comment (flodolo) [inline locales-src/onboarding.ftl]: Per previous conversations, leave this as "Firefox Accounts", don't replace "Firefox".

**Comment by Mardak** [comment]
This is on top of the previous PR #5139 (so tests run correctly). Only the last commit https://github.com/mozilla/activity-stream/pull/5144/commits/b05d3af5cc44c3455bef02d40c3c7e8fa54cb013 is relevant.

**Comment by flodolo** [comment]
> This is on top of the previous PR #5139 (so tests run correctly). Only the last commit [b05d3af](https://github.com/mozilla/activity-stream/commit/b05d3af5cc44c3455bef02d40c3c7e8fa54cb013) is relevant.

So, nothing for me to do here, correct?

**Comment by JeaneC** [comment]
Rebased after #5139

File: content-src/components/DiscoveryStreamComponents/DSLinkMenu/DSLinkMenu.jsx [hunk 2]
```diff
@@ -84,4 +83,4 @@ export class _DSLinkMenu extends React.PureComponent {
   }
 }
 
-export const DSLinkMenu = injectIntl(_DSLinkMenu);
+export const DSLinkMenu = _DSLinkMenu;
```
Comment (punamdahiya) [inline content-src/components/DiscoveryStreamComponents/DSLinkMenu/DSLinkMenu.jsx]: we should not need _DSLinkMenu and can directly export DSLinkMenu class.

File: test/unit/content-src/components/DiscoveryStreamComponents/DSLinkMenu.test.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import {mountWithIntl, shallowWithIntl} from "test/unit/utils";
+import {mount, shallow} from "enzyme";
 import {_DSLinkMenu as DSLinkMenu} from "content-src/components/DiscoveryStreamComponents/DSLinkMenu/DSLinkMenu";
```
Comment (punamdahiya) [inline test/unit/content-src/components/DiscoveryStreamComponents/DSLinkMenu.test.jsx]: we can directly import DSLinkMenu here

**Comment by punamdahiya** [comment]
> > This is on top of the previous PR #5139 (so tests run correctly). Only the last commit [b05d3af](https://github.com/mozilla/activity-stream/commit/b05d3af5cc44c3455bef02d40c3c7e8fa54cb013) is relevant.
> 
> So, nothing for me to do here, correct?

That's correct, no string changes in rebased patch

**Pull request merged**